### PR TITLE
read large zip file

### LINKED ISSUE #92: Read large file from zip
When I try to read a file size greater than 2^31 bytes from a zip archive using ZipFile.jl, I got this error:
```               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.7.2 (2022-02-06)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

(test-zip) pkg> st
      Status `~/test-zip/Project.toml`
  [a5390f91] ZipFile v0.10.0

julia> using ZipFile

julia> zf=ZipFile.Reader("../abc.zip")
ZipFile.Reader for IOStream(<file ../abc.zip>) containing 1 files:

uncompressedsize method  mtime            name
----------------------------------------------
      4604988545 Deflate 2022-08-16 16-58 abc.gif


julia> read(zf.files[1])
ERROR: InexactError: trunc(UInt32, 4604988545)
Stacktrace:
  [1] throw_inexacterror(f::Symbol, #unused#::Type{UInt32}, val::Int64)
    @ Core ./boot.jl:612
  [2] checked_trunc_uint
    @ ./boot.jl:642 [inlined]
  [3] toUInt32
    @ ./boot.jl:726 [inlined]
  [4] UInt32
    @ ./boot.jl:766 [inlined]
  [5] convert
    @ ./number.jl:7 [inlined]
  [6] cconvert
    @ ./essentials.jl:417 [inlined]
  [7] crc32
    @ ~/.julia/packages/ZipFile/6aqG6/src/Zlib.jl:369 [inlined]
  [8] update_reader!(f::ZipFile.ReadableFile, data::Vector{UInt8})
    @ ZipFile ~/.julia/packages/ZipFile/6aqG6/src/ZipFile.jl:517
  [9] read(f::ZipFile.ReadableFile, nb::Int64)
    @ ZipFile ~/.julia/packages/ZipFile/6aqG6/src/ZipFile.jl:561
 [10] read(f::ZipFile.ReadableFile)
    @ ZipFile ~/.julia/packages/ZipFile/6aqG6/src/ZipFile.jl:555
 [11] top-level scope
    @ REPL[4]:1
```

### PR DESCRIPTION
**Pull request by thedacheng**
Read file with size larger than 2^31 from zip. Fix #92

### CODE REVIEW

**Comment by tecosaur** [comment]
I've just run into this myself, and would dearly appreciate a fix. Is there anything stopping this from being mergerd?

**Comment by thedacheng** [comment]
@fhs can you please review this PR? Thanks,

File: src/ZipFile.jl [hunk 1]
```diff
@@ -513,8 +513,15 @@ end
 "advance io position state and crc32 checksum, checking it at eof"
 function update_reader!(f::ReadableFile, data::Array{UInt8})
     f._zpos = position(f._io) - f._datapos
-    f._pos += length(data)
-    f._currentcrc32 = Zlib.crc32(data, f._currentcrc32)
+    datalen = length(data)
+    f._pos += datalen
+    chunk_size = if Sys.WORD_SIZE > 32 2^31 else datalen end 
+    start = 1
+    while datalen > 0
```
Comment (fhs) [inline src/ZipFile.jl:520]: It seems `f._currentcrc32` will be uninitialized if `datalen == 0`, and then we're trying to use it below if `eof(f)`.

Comment (thedacheng) [reply src/ZipFile.jl:520]: `f._currentcrc32` should have already been initialized before this function is called. Please refer to line [https://github.com/fhs/ZipFile.jl/blob/8b2bbf3cc334228489a34cf911754c83309cba78/src/ZipFile.jl#L517](https://github.com/fhs/ZipFile.jl/blob/8b2bbf3cc334228489a34cf911754c83309cba78/src/ZipFile.jl#L517)

Comment (fhs) [reply src/ZipFile.jl:520]: That is the old code. After your change, that line is gone.

Comment (thedacheng) [reply src/ZipFile.jl:520]: Please note `f._currentcrc332` is used as an input in the previous version, which means it has been initialized already by that point. So it should be safe to use it in the new version of the code.

Comment (fhs) [reply src/ZipFile.jl:520]: I see. So, we're not making any new assumption about it being initialized or not.

**Review by fhs — APPROVED** [review]
Thanks!

**Comment by CarloLucibello** [comment]
This is a nice enhancement, @fhs could you tag a new release with it?

**Comment by fhs** [comment]
> This is a nice enhancement, @fhs could you tag a new release with it?

@CarloLucibello You add a comment to the commit to do a release (for example: https://github.com/fhs/ZipFile.jl/commit/a660f9e6705a2674e9afa6a93143b2eb1261d019).
Are you not able to do this as a collaborator?

**Comment by CarloLucibello** [comment]
I'm not a collaborator of this repo (I received the invite for NPZ.jl)

**Comment by fhs** [comment]
@CarloLucibello I just sent an invite for this repo.

**Pull request merged**